Src/FluentAssertions/Specialized/ActionAssertions.cs

```diff
@@ -119,6 +128,14 @@ private Exception InvokeSubjectWithInterception()
             return actualException;
         }
 
+        private void FailIfSubjectAsyncVoid(string assertionMethodName)
+        {
+            if (Subject.GetMethodInfo().HasAttribute<AsyncStateMachineAttribute>())
+            {
+                throw new InvalidOperationException($"Cannot use {assertionMethodName} on an Action representing an async void method.");
```

Comment:
Let me know if you prefer a more descriptive message indicating that Action very likely should have been Func<Task>

Reply:
Yes, I think emphasizing the possible solution does help the user understanding his mistake.